x-pack/plugins/cross_cluster_replication/public/app/sections/home/auto_follow_pattern_list/components/detail_panel/detail_panel.js

```diff
@@ -47,155 +44,154 @@ export class DetailPanel extends Component {
     closeDetailPanel: PropTypes.func.isRequired,
   }
 
-  renderAutoFollowPattern() {
-    const {
-      autoFollowPattern: {
-        followIndexPatternPrefix,
-        followIndexPatternSuffix,
-        remoteCluster,
-        leaderIndexPatterns,
-      },
-    } = this.props;
-
-    let indexManagementFilter;
-    if(followIndexPatternPrefix) {
-      indexManagementFilter = `name:${followIndexPatternPrefix}`;
-    } else if(followIndexPatternSuffix) {
-      indexManagementFilter = `name:${followIndexPatternSuffix}`;
-    }
-    const indexManagementUri = getIndexListUri(indexManagementFilter);
-
+  renderAutoFollowPattern({ followIndexPatternPrefix, followIndexPatternSuffix, remoteCluster, leaderIndexPatterns }) {
```

Comment:
Is there a benefit to requiring arguments? It seems more straightforward to me to destructure `this.props` within the function so it's clear where these variables are coming from.

Reply:
So here it's purely preferences. The huge benefit of functional programming is this one: no need to ask or wonder where the arguments of a function come from (it is not its responsibility to know that). Simply write a function, give it an input, and expect an output.

The benefits?

- If it was a complex function, it could be exported and tested very easily
- If the component starts to grow unexpectedly, we'd cut and paste those functions in separate files and we already have our new Function components ready and working.

The mental model obviously has to change when reading the code. The question is not anymore: "but where do those arguments come from?", but: "does it return the correct jsx with the arguments that have been provided?".